incubator/hnc/pkg/kubectl/root.go

```diff
@@ -36,10 +36,24 @@ import (
 var k8sClient *kubernetes.Clientset
 var hncClient *rest.RESTClient
 var rootCmd *cobra.Command
+var apiServer *APIServer
+
+type APIServer struct{}
+
+type hncAPIServer interface {
+	getHierarchy(nnm string) *api.HierarchyConfiguration
+	updateHierarchy(hier *api.HierarchyConfiguration, reason string)
+}
+
+func NewAPIServer() *APIServer {
+	return &APIServer{}
+}
```

Comment:
A few things here:

- I think I suggested calling this type "Server" but it's really a client (see `k8sClient` and `hncClient`, above), so let's change the name to client.
- The variable should have the _interface_ type, not the _concrete_ type. Otherwise we can't replace the real server with a fake one.
- I'd suggest calling the interface type `Client` (or maybe `APIClient`) and the real implementation `realClient`. The interface is the type everyone will be using, so give it the better name.
- There's no need to have a method called `NewClient`. In `init()`, simply say `client = &realClient{}`. As far as I've seen, that's the preferred Go style (avoid a `New` function) unless there's nontrivial processing required to create an object.

Reply:
Finally, if `k8sClient` and `hncClient` are no longer used anywhere _except_ in `realClient`, maybe put them into that struct so they're not globals anymore?